core/src/main/java/org/elasticsearch/cluster/routing/PlainShardsIterator.java

```diff
@@ -74,7 +75,12 @@ public int sizeActive() {
     }
 
     @Override
-    public Iterable<ShardRouting> asUnordered() {
+    public List<ShardRouting> getShardRoutings() {
         return shards;
```

Comment:
Maybe wrap it in an unmodifiable list in the ctor? I think we should not allow mutation here?